yt_dlp/extractor/mainstreaming.py

```diff
@@ -0,0 +1,222 @@
+        return host
+
+    def _get_webtools_base_url(self, host):
+        return f'{self.http_scheme()}//{self._get_webtools_host(host)}'
+
+    def _call_api(self, host: str, path: str, item_id: str, query=None, note='Downloading API JSON', fatal=False):
+        # JSON API, does not appear to be documented
+        return self._call_webtools_api(host, '/api/v2/' + path, item_id, query, note, fatal)
+
+    def _call_webtools_api(self, host: str, path: str, item_id: str, query=None, note='Downloading webtools API JSON', fatal=False):
+        # webtools docs: https://webtools.msvdn.net/
+        return self._download_json(
+            urljoin(self._get_webtools_base_url(host), path), item_id, query=query, note=note, fatal=fatal)
+
+    def _real_extract(self, url):
+        host, video_id = self._match_valid_url(url).groups()
+        content_info = try_get(
+            self._call_api(
+                host, f'content/{video_id}', video_id, note='Downloading content info API JSON'), lambda x: x['playerContentInfo'])
+        # Fallback
+        if not content_info:
+            webpage = self._download_webpage(url, video_id)
+            player_config = self._parse_json(
+                self._search_regex(
+                    r'config\s*=\s*({.+?})\s*;', webpage, 'mainstreaming player config',
+                    default='{}', flags=re.DOTALL),
+                video_id, transform_source=js_to_json, fatal=False) or {}
+            content_info = player_config['contentInfo']
+
+        host = content_info.get('host') or host
+        video_id = content_info.get('contentID') or video_id
+        title = content_info.get('title')
+        description = traverse_obj(
+            content_info, 'longDescription', 'shortDescription', expected_type=str, get_all=False)
+        is_live = was_live = False
+        if content_info.get('drmEnabled'):
+            self.report_drm(video_id)
+
+        alternative_content_id = content_info.get('alternativeContentID')
+        if alternative_content_id:
+            self.report_warning(f'Ignoring alternative content ID: {alternative_content_id}')
+
+        content_type = int_or_none(content_info.get('contentType'))
+        format_base_url = None
+        formats = []
+        subtitles = {}
+        # Live content
+        if content_type == 20:
+            dvr_enabled = traverse_obj(content_info, ('playerSettings', 'dvrEnabled'), expected_type=bool)
+            format_base_url = f"https://{host}/live/{content_info['liveSourceID']}/{video_id}/%s{'?DVR' if dvr_enabled else ''}"
+            is_live = True
+            heartbeat = self._call_api(host, f'heartbeat/{video_id}', video_id, note='Checking stream status') or {}
+            if heartbeat.get('heartBeatUp') is False:
+                self.raise_no_formats(f'MainStreaming said: {heartbeat.get("responseMessage")}', expected=True)
+                is_live = False
+                was_live = True
```

Comment:
```suggestion
                live_status = 'was_live'
```
similar elsewhere would be simpler. You can also pass this directly in the infodict